.travis.yml

```diff
@@ -27,156 +27,185 @@ env:
-        PATH="/usr/local/opt/ccache/bin:$PATH"
-        CMAKE_BUILD_PARALLEL_LEVEL=4
-        CTEST_PARALLEL_LEVEL=1
-      before_install:
-        # Print some information about CMake and ccache
-        - cmake --version
-        - ccache -s
-        # Download and prepare our build environment
-        - mkdir "${MIXXX_ENVPATH}"
-        - read -r MIXXX_ENVNAME < build/osx/golden_environment
-        - curl "https://downloads.mixxx.org/builds/buildserver/2.3.x-macosx/${MIXXX_ENVNAME}.tar.gz" | tar xzf -  --strip 1 -C "${MIXXX_ENVPATH}"
-        # FIXME: This fixes some wrong hardcoded paths inside pkg-config files of the build environment (e.g. in "lib/pkgconfig/taglib.pc")
-        # This will only work if neither MIXXX_ENVNAME nor MIXXX_ENVPATH contain special regex characters!
-        - find "${MIXXX_ENVPATH}" -name "*.pc" -or -path "*/bin/taglib-config" -exec sed -i".orig" -e "s|/Users/mixxx/bs-2.3-mac/amd64/environment/${MIXXX_ENVNAME}|${MIXXX_ENVPATH}|g" {} \;
-        - export QT_DIR="$(find "${MIXXX_ENVPATH}" -type d -path "*/cmake/Qt5")"
-        - export QT_QPA_PLATFORM_PLUGIN_PATH="$(find "${MIXXX_ENVPATH}" -type d -path "*/plugins")"
-      install:
-        - mkdir cmake_build
-        - cd cmake_build
-        - cmake -L $CMAKEFLAGS $CMAKEFLAGS_EXTRA -DCMAKE_PREFIX_PATH=${MIXXX_ENVPATH} -DQt5_DIR=${QT_DIR} ..
-        - cmake --build .
-        - cpack -G DragNDrop
+        - $HOME/nix.store
+        - $HOME/.ccache
+      before_install: &nix_before_install
+        - sudo mkdir -p /etc/nix
+        - echo "substituters = https://cache.nixos.org/ file://$HOME/nix.store" | sudo tee -a /etc/nix/nix.conf > /dev/null
+        - echo 'require-sigs = false' | sudo tee -a /etc/nix/nix.conf > /dev/null
+        - sudo mkdir -p /etc/nix && echo 'sandbox = true' | sudo tee /etc/nix/nix.conf
+        - mkdir -p ~/.config/nixpkgs; echo '{ allowBroken = true; }' > ~/.config/nixpkgs/config.nix
+      before_cache: &nix_before_cache
+        - mkdir -p $HOME/nix.store
+        - nix copy --to file://$HOME/nix.store -f default.nix buildInputs
+      script:
+        - nix-build shell.nix --arg releaseMode true --arg defaultLv2Plugins true
+    - name: Nix development shell build
+      install: *nixinstall
+      cache: *nixcache
+      before_cache: *nix_before_cache
+      before_install: *nix_before_install
       script:
-        # Run tests and benchmarks
-        - ctest --timeout 45
-        - cmake --build . --target benchmark
-      before_cache:
-        # Avoid indefinite cache growth
-        - brew cleanup
-        # Cache only .git files under "/usr/local/Homebrew" so "brew update"
-        # does not take 5min every build
-        # Source: https://discourse.brew.sh/t/best-practice-for-homebrew-on-travis-brew-update-is-5min-to-build-time/5215/12
-        - find /usr/local/Homebrew \! -regex ".+\.git.+" -delete
+        - nix-shell --arg enableKeyfinder true --arg defaultLv2Plugins true --command "configure; build;"
```

Comment:
Did you do this stuff in order to test you PR? Maybe we should merge a separate PR that adds Nix CI first.

Reply:
The current file does not even build on travis. I'm working on it now to get it into a state where it works in both modes